cmd/minishift/cmd/start.go

```diff
@@ -145,12 +141,18 @@ func runStart(cmd *cobra.Command, args []string) {
 	defer libMachineClient.Close()
 
 	validateOpenshiftVersion()
-	validateProxyArgs()
 
-	setDockerProxy()
-	setOcProxy()
-	setShellProxy()
-	setSubcriptionManagerParameters()
+	proxyConfig, err := util.NewProxyConfig(viper.GetString(httpProxy), viper.GetString(httpsProxy), viper.GetString(noProxyList))
```

Comment:
Could this be moved to a separate function? We keep stuffing everything into the `runStart` and this function is becoming a monster. This can be handled outside of #409.